.github/actions/git_checkout_from_cache/action.yml

```diff
@@ -0,0 +1,34 @@
+name: git_checkout_from_cache
+description: Checkout git from cache
+runs:
+  using: composite
+  steps:
+  - name: restore_cache
+    uses: actions/cache@v3.3.1
+    with:
+      key: source-v{{ .Environment.CACHE_VERSION }}-{{ .Branch }}-{{ .Revision }}
+      path: UPDATE_ME
+      restore-keys: |-
+        source-v{{ .Environment.CACHE_VERSION }}-{{ .Branch }}-{{ .Revision }}
+        source-v{{ .Environment.CACHE_VERSION }}-{{ .Branch }}-
+        source-v{{ .Environment.CACHE_VERSION }}-
+  - name: Fetch git tags
+    run: |-
+      mkdir -p ~/.ssh
+      ssh-keyscan github.com >> ~/.ssh/known_hosts
+      echo "github.com ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ==" >> ~/.ssh/known_hosts
```

Comment:
should we store this in secrets 👀

Reply:
yes, you are right.
i copied it from the old circleci config that we have, its there already.
anyways i will do it right away

Reply:
We probably need to recycle this key too!

Reply:
We don't need this line. ssh-keyscan at line 18 will get the public key of github.com and store it in known_hosts.

Reply:
thanks @balakrishna-deriv for the confirmation
I've deleted the line